src/main/java/bio/terra/app/controller/UnauthenticatedApiController.java

```diff
@@ -59,9 +66,24 @@ public UnauthenticatedApiController(
 
     @Override
     public ResponseEntity<RepositoryConfigurationModel> retrieveRepositoryConfig() {
+        String semVer = "1.0.0-UNKNOWN";
+        String gitHash = "00000000";
+        try {
+            Properties properties = new Properties();
+            InputStream versionFile = getClass().getClassLoader().getResourceAsStream("version.properties");
```

Comment:
This looks like one of those stream things that we need to make sure gets closed by using a resource try.

I and concerned that we are reading the file every time somebody hits the end point. Especially because it is unauthenticated. What would be the consequence of storing the result in a static only doing this read once?

Reply:
Good point about streams. I'll read the version data once and store it in a static instead.